File: public/examples/integrations/mcp-servers/stripe_api/list_payment_method_configurations_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "StripeApi.ListPaymentMethodConfigurations";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "filter_by_connect_application": "ca_1Gqj2XExample",
  "pagination_ending_before_id": "pmc_001122334455",
  "expand_fields": [
    "data.default_price",
    "data.supported_currencies"
  ],
  "max_results": 25,
  "pagination_starting_after_id": "pmc_556677889900"
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

File: public/examples/integrations/mcp-servers/stripe_api/list_payment_method_configurations_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "StripeApi.ListPaymentMethodConfigurations"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'filter_by_connect_application': 'ca_1Gqj2XExample',
    'pagination_ending_before_id': 'pmc_001122334455',
    'expand_fields': ['data.default_price', 'data.supported_currencies'],
    'max_results': 25,
    'pagination_starting_after_id': 'pmc_556677889900'
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))
